src/Microsoft.AspNetCore.SignalR.Client/RpcConnection.cs

```diff
@@ -127,6 +180,38 @@ private string GetNextId()
             return (Interlocked.Increment(ref _nextId) - 1).ToString();
         }
 
+        Type IInvocationBinder.GetReturnType(string invocationId)
```

Comment:
this explicit interface implementation is quite ugly - can we have a separate internal/nested class we pass as IInvocationBinder instead of RpcConnection?